pickle_batch_loader.py

```diff
@@ -0,0 +1,38 @@
+""" For loading random batches of n objects from a pickle file without taking up much memory at any one time """
+import os
+import pickle
+import random
+
+def load_batch(filepath, batch_size):
+    """ Returns a specified number of objects at random from a pickle file 
+    
+    No more than (batch_size + 1) objects required to be in memory at once."""
+    # First we need to find how many pickled objects there are in the file
+    # I don't know any more efficient way to do this than to just load and discard every object in the file
+    obj_count = 0
+    while True:
+        try:
+            pickle.load(open(filepath, mode='rb'))
```

Comment:
I'm pretty sure that you can cut out the mode= part. 'rb' by itself should work the same.